sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalStorageTelemetryPipelineListener.java

```diff
@@ -46,16 +58,40 @@ public LocalStorageTelemetryPipelineListener(
 
     @Override
     public void onResponse(TelemetryPipelineRequest request, TelemetryPipelineResponse response) {
-        if (StatusCode.isRetryable(response.getStatusCode())) {
+        int statusCode = response.getStatusCode();
+        if (StatusCode.isRetryable(statusCode)) {
             localFileWriter.writeToDisk(
-                request.getConnectionString(), request.getTelemetry(), getOriginalErrorMessage(response));
+                request.getConnectionString(), request.getByteBuffers(), getOriginalErrorMessage(response));
+        } else if (statusCode == 206) {
+            processStatusCode206(request, response);
+        }
+    }
+
+    private void processStatusCode206(TelemetryPipelineRequest request, TelemetryPipelineResponse response) {
+        Set<ResponseError> errors = response.getErrors();
+        errors.forEach(error -> logger.verbose("Error in telemetry: " + error));
```

Comment:
(to avoid the unnecessary allocations when verbose logging is not enabled)
```suggestion
        errors.forEach(error -> logger.verbose("Error in telemetry: {}", error));
```

and/or surround in if verbose enabled conditional